sdk/storage/Azure.Storage.DataMovement/src/JobPlan/JobPartPlanFile.cs

```diff
@@ -65,8 +65,11 @@ public static async Task<JobPartPlanFile> CreateJobPartPlanFileAsync(
             try
             {
                 using (FileStream fileStream = File.Create(result.FileName.ToString()))
+                using (MemoryStream ms = new())
                 {
-                    await headerStream.CopyToAsync(fileStream, DefaultBufferSize, cancellationToken).ConfigureAwait(false);
+                    header.Serialize(ms);
+                    ms.Position = 0;
+                    await ms.CopyToAsync(fileStream, DefaultBufferSize, cancellationToken).ConfigureAwait(false);
                 }
```

Comment:
This is not great but is also exactly what the code was doing before elsewhere. All that's changed is it's moved closer to the actual file stream write.

In the future we shouldn't be using a memory stream like this for such a small piece of data. It's extra copies just to move the data, plus all the underlying array resizing of the stream. We don't want that much memory churn in an already memory-intensive product.

Realistically we move to renting a single array that is sized out ahead of time and directly write that array to the file. Async serialization directly to file seems undesirable in this scenario due to lots of single byte writes.

Reply:
@jalauzon-msft I think your changing the same lines of code and there might be a merge conflict